quipucords/scanner/network/inspect.py

```diff
@@ -297,6 +303,89 @@ def _inspect_scan(  # noqa: PLR0912, PLR0915, C901
             self.scan_task.log_message(log_message, log_level=logging.DEBUG)
         return error_msg, scan_result
 
+    @transaction.atomic
+    def _persist_results(self, ansible_results: AnsibleResults):
+        facts = self._post_process_facts(ansible_results)
+        self.scan_task.log_message(
+            f"host scan complete for {ansible_results.host}."
+            f" Status: {ansible_results.status}. Facts {facts}",
+            log_level=logging.DEBUG,
+        )
+        sys_result = SystemInspectionResult(
+            name=ansible_results.host,
+            status=ansible_results.status,
+            source=self.scan_task.source,
+            task_inspection_result=self.scan_task.inspection_result,
+        )
+        sys_result.save()
+        raw_facts = []
+        for fact_key, fact_value in facts.items():
+            raw_facts.append(
+                RawFact(
+                    name=fact_key,
+                    value=fact_value,
+                    system_inspection_result=sys_result,
+                )
+            )
+        RawFact.objects.bulk_create(raw_facts)
+        increment_kwargs = self._get_scan_task_increment_kwargs(ansible_results.status)
+        self.scan_task.increment_stats(ansible_results.host, **increment_kwargs)
+
+    def _post_process_facts(self, ansible_results):
+        logger.debug(
+            "[host=%s] post processing facts. Unprocessed facts=%s",
+            ansible_results.host,
+            ansible_results.facts,
+        )
+        facts = {}
+        for fact_key, fact_value in ansible_results.facts.items():
+            try:
+                processed_fact = process(
+                    scan_task=self.scan_task,
+                    previous_host_facts=facts,
+                    fact_key=fact_key,
+                    fact_value=fact_value,
+                    host=ansible_results.host,
+                )
+            except Exception:
+                logger.exception(
+                    "[host=%s] Unexpected error ocurred during fact '%s' processing",
+                    ansible_results.host,
+                    fact_key,
+                )
+                continue
```

Comment:
in the original implementation this code wasn't wrapped in try/except.

this is probably overkill, but I'm still unsure on how well tested and resilient `process` is